### PR TITLE
Make types like `Box<dyn DynAccess>` implement Access

### PR DESCRIPTION
**Pull request by steffahn**
by generalizing the P: Deref-implementation to ?Sized,
and implementing Access for some `dyn DynAccess` trait object types.

Not yet documented or polished, this PRs main purpose *for now* is to allow CI testing.

### CODE REVIEW

**Comment by steffahn** [comment]
The failing clippy lints are unrelated to this PR.

**Comment by steffahn** [comment]
Older compiler version apparently fails in one of the dependencies (`half`).

**Comment by steffahn** [comment]
@vorner I’ve addressed the unrelated CI failures (I don’t know if that’s the "proper" way to do this with the dependencies for 1.45 compilation, but it seems like it works); feel free to re-run CI.

File: src/access.rs [hunk 1]
```diff
@@ -112,13 +112,53 @@ pub trait Access<T> {
...
+
+impl<T> Access<T> for dyn DynAccess<T> + '_ + Sync + Send {
+    type Guard = DynGuard<T>;
+
+    fn load(&self) -> Self::Guard {
+        self.load()
+    }
+}
+
+// Should probably be moved to a more appropriate place / perhaps more extensively tested / etc
```
Comment (vorner) [inline src/access.rs:146]: The more appropriate place is just the end of the file (not the middle), otherwise it is fine :-)

**Review by vorner — COMMENTED** [review]
Thanks for the provided fixes. There's also some miri, but I think that one should be dealt with separately… it's a new miri's capability that wasn't there before and needs some more investigation.

Seems like it indeed helps and it can be compiled on the ancient versions. If you also could adjust the documentation, stating that the wrapper is not necessary any more in many cases (are there any cases left where it is necessary? It needs to be left in place for backwards compatibility, but maybe if there's no use case for it any more, it could be deprecated and hidden from documentation).

**Comment by steffahn** [comment]
> are there any cases left where it is necessary?

as already noted over in #77:

> the actual `AccessConvert` is somewhat more general, as it allows – say – `Box<dyn Foo>` for a trait `Foo: DynAccess`, too.

Similarly, you can use it for things like `Box<dyn DynAccess + Unpin>` or `Box<dyn DynAccess + Sync>` or other more rare variantions on the trait object type that I didn't cover with any explicit implementation. (I chose the same selection three of auto-trait combinations (none, only send, and send+sync) that e. g. `dyn Any` in the standard library supports for downcasting; if there are others with practical use, they can be added later.)

**Comment by vorner** [comment]
Hello again. Just wanted to check, are you still working on the docs, or did that one get lost in the rest of the discussion? 😇 Just that we both don't wait on something from the other side.

**Comment by steffahn** [comment]
Thanks for the ping. I didn’t misunderstand, but I haven’t written the docs yet either; I’ll try to allocate some time to this soon :-)

**Comment by vorner** [comment]
Hello.

I've discovered this still sits here. Not that I would have too much free time myself, but should I just take it over and write the docs, so it moves forward?

**Comment by steffahn** [comment]
I forgot this PR even exists, so ehm... sure, feel free to take over, thank you.

**Comment by vorner** [comment]
I've taken the main commit, put it onto current (which already had the CI issues mostly solved) master and made some docs for it and merged in #85.

I'll try to collect few more outstanding PRs and release a new version soon. If I don't get around to it by the end of the year, feel free to ping me.

Thank you for discovering the way to do this :-)

**Pull request closed without merging**